ts/localization/localeTools.ts

```diff
@@ -106,6 +106,17 @@ type ArgsFromTokenStr<T extends SimpleLocalizerTokens | PluralLocalizerTokens> =
 
 export type ArgsFromToken<T extends MergedLocalizerTokens> = MappedToTsTypes<ArgsFromTokenStr<T>>;
 
+export type ArgsFromTokenWithIcon<
+  T extends MergedLocalizerTokens,
+  I extends string = string,
+> = MappedToTsTypes<ArgsFromTokenStr<T>> & { icon: I };
+
+export function isArgsFromTokenWithIcon<T extends MergedLocalizerTokens, I extends string>(
+  args: ArgsFromToken<T> | undefined
+): args is ArgsFromTokenWithIcon<T, I> {
+  return !!args && !isEmptyObject(args) && 'icon' in args && typeof args.icon === 'string';
+}
```

Comment:
I am not sure this is correct: 
the return test will be true when args.icon is a string, 
`typeof args.icon === 'string';` 
but the type guard is making is `args is ArgsFromTokenWithIcon<T, I>` where `I extends string`, i .e. can be more specific. 

Or am I misleaded?

Reply:
I think the only way to make this work is to bring the type of allowed lucide icons in this file.